impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js

```diff
@@ -136,24 +156,29 @@ define([
         orient: orient,
         left: left,
         top: top,
-        backgroundColor: legendBgColor.toLowerCase() !== "#ffffff" ? legendBgColor : "transparent",
+        backgroundColor: legendBgColor,
         itemWidth: 8,
         itemHeight: 8,
         padding: 10,
+        tooltip: {
+          show: true
+        },
         textStyle: {
-          fontWeight: fontWeight === "plain" ? "normal" : fontWeight,
-          color: model.legendColor,
-          fontSize: model.legendSize,
-          fontFamily: fontFamily === "Default" ? font.substring(font.indexOf(" ") + 1) : fontFamily
+          width: 80,
+          overflow: "truncate",
```

Comment:
To prevent legend cover most of the chart when `fontSize` is increased and instead truncate the labels in legend and show the full name when hovered.